extensions/stackdriver/common/context.cc

```diff
@@ -0,0 +1,44 @@
+/* Copyright 2019 Istio Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "extensions/stackdriver/common/context.h"
+#include "extensions/stackdriver/common/constants.h"
+#include "google/protobuf/util/json_util.h"
+
+namespace Extensions {
+namespace Stackdriver {
+namespace Common {
+
+using namespace google::protobuf::util;
+using namespace stackdriver::common;
+
+Status ExtractNodeMetadata(const google::protobuf::Struct &metadata,
```

Comment:
naming: `extractNodeMetadata`

Reply:
IIRC envoy replace all `google::protobuf` to `ProtobufWkt`. Not sure why but try if you can follow

Reply:
Not necessary. This is done to simplify copybara import (protobuf is net::proto2 internally). We are not planning to move this into google3 any time soon.

Reply:
Rename the function to be lower cased. I don't want to introduce dependency on envoy code base because eventually it will be compiled into a wasm module, which ideally should not rely on anything from envoy except the SDK.

Reply:
I see. Thank you for the explanation!